### ISSUE TITLE
How to Deploy a Single Instance of yolov8_ros Node for Multiple Camera Detections

### ISSUE THREAD

**Issue by Ekanshh** [mgonzs13/yolo_ros#20]
I am currently utilizing the yolov8_ros node for object detections and tracking on camera streams. However, the current setup involves deploying a separate yolov8_ros node for each camera, leading to performance issues. I am interested in exploring the possibility of deploying a single instance of the yolov8_ros node to handle detections on multiple cameras concurrently. 

Here is a brief overview of the current vs desired process:

- Current System: 
![image](https://github.com/mgonzs13/yolov8_ros/assets/38471801/3c3996dc-c771-45b9-9e97-7c875994c843)

- Desired System:
![image](https://github.com/mgonzs13/yolov8_ros/assets/38471801/9d2cf7cc-9c19-45cd-bcca-8c783d00b4b0)

**Comment by mgonzs13** [comment]
Hi @Ekanshh, what about creating a node that subscribes to multiple camera topics and then publishes those images in the input topic of the yolov8_node?

**Comment by Ekanshh** [comment]
Thank you for your quick response. I will give it a try and provide an update here to let you know how it goes.

**Comment by RP-ND** [comment]
Did you ever manage to go further with this @Ekanshh?

**Comment by Ekanshh** [comment]
> Did you ever manage to go further with this @Ekanshh?

No, I haven’t made any progress yet. I’m still sticking to separate nodes. Essentially, I’ve customized the bringup launch file to accommodate 6 nodes for two cameras, with 2 nodes dedicated to detection, 2 for tracking, and 2 for debugging purposes. However, it’s currently on my to-do list and not of high priority at the moment.

**Comment by RP-ND** [comment]
> > Did you ever manage to go further with this @Ekanshh?
> 
> No, I haven’t made any progress yet. I’m still sticking to separate nodes. Essentially, I’ve customized the bringup launch file to accommodate 6 nodes for two cameras, with 2 nodes dedicated to detection, 2 for tracking, and 2 for debugging purposes. However, it’s currently on my to-do list and not of high priority at the moment.

Fair, I have an embedded front end with a Nvidia Orin Nano, I may go further with this since running two custom nodes, even if they are 8n, reaches the compute limit of the frontend memory wise.

**Comment by Ekanshh** [comment]
> > > Did you ever manage to go further with this @Ekanshh?
> > 
> > 
> > No, I haven’t made any progress yet. I’m still sticking to separate nodes. Essentially, I’ve customized the bringup launch file to accommodate 6 nodes for two cameras, with 2 nodes dedicated to detection, 2 for tracking, and 2 for debugging purposes. However, it’s currently on my to-do list and not of high priority at the moment.
> 
> Fair, I have an embedded front end with an Orin Nano, I may go further with this since running two custom nodes, even if they are 8n, reaches the compute limit of the frontend memory wise.

If you do come up with a working solution or make further progress, let me know. :)

**Comment by Ekanshh** [comment]
Closing this issue as I've successfully resolved it for my particular scenario by implementing the suggestion provided by @mgonzs13. My approach involved modifying the yolov8_ros node to subscribe to multiple camera topics synchronously and utilizing YOLO to batch predict on these images. Subsequently, I forwarded the predictions to multiple tracking topics corresponding to the camera topics.

**Comment by Edwardius** [comment]
hey @mgonzs13 , would you be open to me helping out with integrating some sort of multi-camera handling for this?

Similar to what @Ekanshh here did with having a synchronization layer followed by a batch prediction process. If you have ideas of your own for how to implement, let me know.

**Comment by mgonzs13** [comment]
Hi @Edwardius, do you want to synchronize all images from each camera and then apply YOLO?
